src/instructlab/configuration.py

```diff
@@ -390,48 +580,46 @@ def __init__(self, config_obj: Config):
-            )
-        else:
-            try:
-                config_obj = read_config(config_file)
-            except ConfigException as ex:
-                raise click.ClickException(str(ex))
-        # setup logging
-        log.configure_logging(log_level=config_obj.general.log_level.upper())
-        ctx.obj = Lab(config_obj)
-        # default_map holds a dictionary with default values for each command parameters
-        config_dict = get_dict(ctx.obj.config)
-        # since torch and train args are sep, they need to be combined into a single `train` entity for the default map
-        # this is because the flags for `ilab model train` will only be populated if the default map has a single `train` entry, not two.
-        config_dict["train"] = (
-            config_dict["train"]["train_args"]
-            | config_dict["train"]["torch_args"]
-            | config_dict["train"]["train_args"]["lora"]
-            | config_dict["train"]["train_args"]["deepspeed_options"]
-        )
-        config_dict["evaluate"] = (
-            config_dict["evaluate"]
-            | config_dict["evaluate"]["mmlu"]
-            | config_dict["evaluate"]["mmlu_branch"]
-            | config_dict["evaluate"]["mt_bench"]
-            | config_dict["evaluate"]["mt_bench_branch"]
-        )
-        # need to delete the individual sub-classes from the map
-        del config_dict["evaluate"]["mmlu"]
-        del config_dict["evaluate"]["mmlu_branch"]
-        del config_dict["evaluate"]["mt_bench"]
-        del config_dict["evaluate"]["mt_bench_branch"]
+def init(ctx, config_file: str):
+    if ctx.invoked_subcommand in ["init", "sysinfo"] or "--help" in sys.argv[1:]:
+        # do nothing
+        return
 
-        ctx.default_map = config_dict
+    if config_file == "DEFAULT":
+        config_obj = get_default_config()
+    elif not os.path.isfile(config_file):
+        config_obj = None
+        ctx.fail(
+            f"`{config_file}` does not exists, please run `ilab init` "
+            "or point to a valid configuration file using `--config=<path>`."
+        )
+    else:
+        try:
+            config_obj = read_config(config_file)
+        except ConfigException as ex:
+            raise click.ClickException(str(ex))
+    # setup logging
+    log.configure_logging(log_level=config_obj.general.log_level.upper())
+    ctx.obj = Lab(config_obj)
+    # default_map holds a dictionary with default values for each command parameters
+    config_dict = get_dict(ctx.obj.config)
+    # since torch and train args are sep, they need to be combined into a single `train` entity for the default map
+    # this is because the flags for `ilab model train` will only be populated if the default map has a single `train` entry, not two.
+
+    # XXX(osilkin): the following block of code is to hack around conflicting keys
+    #               being merged and erasing each other
+    #               issue: https://github.com/instructlab/instructlab/issues/1523
```

Comment:
PR #1591 will solve this mess and get rid of config_dict flattening. The new code in the PR allows to specify where a value comes from, e.g. `config_section=train_args.lora` to look up values from `train.train_args.lora` in the `train` callback.

Reply:
resolved by #1591